dask_ml/decomposition/pca.py

```diff
@@ -396,6 +408,8 @@ def fit_transform(self, X, y=None):
 
         """
         # X = check_array(X)
+        if isinstance(X, (np.ndarray, pd.DataFrame)):
```

Comment:
Sorry for not seeing this earlier, but can you replace this with `not dask.is_dask_collection(X)`? That will be a bit nicer w.r.t. any duck typing that scikit-learn may do (I don't know if a scipy.sparse matrix is allowed by scikit-learn here. If not, I'd let them raise on that.)

Reply:
Same comment on L200.

Reply:
Thanks. There was actually a test for sparse inputs, and some code to raise an error if sparse inputs passed. I made slight modifications to existing tests:

* `test_pca_sparse_input`: added an argument to make sure the right `TypeError` is raised
* `test_pca_bad_solver`: passed in Dask array instead of NumPy array to make solver-checking code could be reached.